Adyen/Model/Amount.swift

```diff
@@ -119,3 +115,54 @@ extension Amount: Comparable {
     }
 
 }
+
+/// :nodoc:
+public struct AmountComponents {
+    
+    /// :nodoc:
+    public init(amount: Amount) {
+        if let comps = Self.extractAmountComponents(from: amount.formatted) {
+            (self.formattedCurrencySymbol, self.formattedValue) = comps
+        } else {
+            self.formattedValue = String(amount.value)
+            self.formattedCurrencySymbol = amount.currencyCode
+        }
+    }
```

Comment:
maybe the initializer can be internal?, since merchants doesn't need to use the initializer.

Reply:
yes, you are right. it can even be `fileprivate`